frontend/app/controllers/spree/orders_controller.rb

```diff
@@ -42,6 +42,10 @@ def edit
       @order = current_order || Spree::Order.incomplete.find_or_initialize_by(guest_token: cookies.signed[:guest_token])
       authorize! :read, @order, cookies.signed[:guest_token]
       associate_user
+      # It's possible another error may already be assigned so we must join them.
```

Comment:
I think we should also redirect the user when this happens if possible. Allowing to change the current order presenting an old order's URL could be confusing for users.

If possible, what about doing that right after the `@order` assignment on line 42? This way, we can maybe even ignore merging the error message with others?

What do you think?

Reply:
@kennyadsl I think the redirect definitely makes sense so I'll do that, but I don't think I can drop the error message merging without losing one of the messages along the way although it's worth a shot.

Reply:
@kennyadsl I think the redirect definitely makes sense so I'll do that, but I don't think I can drop the error message merging without losing one of the messages along the way although it's worth a shot.

Reply:
Do you know which kind of other messages could we have there?

Reply:
When a product becomes unavailable, and is removed from your cart.

Reply:
When the product becomes unavailable we are redirected to the cart with another flash message but we should not see the new `cannot_edit_completed_orders` flash, right?

I think that this is happening because we are not checking the existence of the `params[:id]` in the if statement. We should print the error message only when `@order.number != params[:id]` and `params[:id]` is set otherwise, as it is now, it will be always be displayed in the cart page.

I think that we can get rid of the messages join if we do this and redirect the user to `spree.order_path(params[:id])`.

Reply:
@kennyadsl updated 👍